loopy/statistics.py

```diff
@@ -983,97 +983,104 @@ def map_product(self, expr):
                               kernel_name=self.knl.name): self.one}) \
-                                + self.rec(expr.numerator) \
-                                + self.rec(expr.denominator)
+                                + self.rec(expr.numerator, tags) \
+                                + self.rec(expr.denominator, tags)
 
     map_floor_div = map_quotient
     map_remainder = map_quotient
 
-    def map_power(self, expr):
+    def map_power(self, expr, tags):
         return self.new_poly_map({Op(dtype=self.type_inf(expr),
                               name="pow",
+                              tags=tags,
                               count_granularity=self.arithmetic_count_granularity,
                               kernel_name=self.knl.name): self.one}) \
-                                + self.rec(expr.base) \
-                                + self.rec(expr.exponent)
+                                + self.rec(expr.base, tags) \
+                                + self.rec(expr.exponent, tags)
 
-    def map_left_shift(self, expr):
+    def map_left_shift(self, expr, tags):
         return self.new_poly_map({Op(dtype=self.type_inf(expr),
                               name="shift",
+                              tags=tags,
                               count_granularity=self.arithmetic_count_granularity,
                               kernel_name=self.knl.name): self.one}) \
-                                + self.rec(expr.shiftee) \
-                                + self.rec(expr.shift)
+                                + self.rec(expr.shiftee, tags) \
+                                + self.rec(expr.shift, tags)
 
     map_right_shift = map_left_shift
 
-    def map_bitwise_not(self, expr):
+    def map_bitwise_not(self, expr, tags):
         return self.new_poly_map({Op(dtype=self.type_inf(expr),
                               name="bw",
+                              tags=tags,
                               count_granularity=self.arithmetic_count_granularity,
                               kernel_name=self.knl.name): self.one}) \
-                                + self.rec(expr.child)
+                                + self.rec(expr.child, tags)
 
-    def map_bitwise_or(self, expr):
+    def map_bitwise_or(self, expr, tags):
         return self.new_poly_map({Op(dtype=self.type_inf(expr),
                               name="bw",
+                              tags=tags,
                               count_granularity=self.arithmetic_count_granularity,
                               kernel_name=self.knl.name):
                            self.zero + (len(expr.children)-1)}) \
-                                + sum(self.rec(child) for child in expr.children)
+                              + sum(self.rec(child, tags) for child in expr.children)
 
     map_bitwise_xor = map_bitwise_or
     map_bitwise_and = map_bitwise_or
 
-    def map_if(self, expr):
+    def map_if(self, expr, *args):
```

Comment:
Why switch to `*args` here?

Reply:
In [0bbbaec](https://github.com/inducer/loopy/pull/688/commits/0bbbaecf30221d6bcdbc89f81cf32fb13c29c9ba), I changed all occurrences to `tags` with a default argument. Does this look reasonable?